lib/unstrctrd.ml

```diff
@@ -88,45 +88,37 @@ let without_comments lst =
       ( match Uchar.to_int uchar with
         | 0x22 (* '"' *) ->
           ( match escaped, quoted_string, stack with
-          | true,  true,  0 -> go stack ~escaped:false ~quoted_string (value :: acc) r
-          | true,  false, 0 -> go stack ~escaped:false ~quoted_string (value :: acc) r
-          | false, true,  0 -> go stack ~escaped ~quoted_string:false (value :: acc) r
-          | false, false, 0 -> go stack ~escaped:false ~quoted_string:true (value :: acc) r
-          | true,  true,  _ -> go stack ~escaped:false ~quoted_string acc r
-          | true,  false, _ -> go stack ~escaped:false ~quoted_string acc r
-          | false, true,  _ -> go stack ~escaped ~quoted_string:false acc r
-          | false, false, _ -> go stack ~escaped ~quoted_string:true  acc r )
+          | true, _,  0 -> go stack ~escaped:false ~quoted_string (value :: acc) r
+          | true, _,  _ -> go stack ~escaped:false ~quoted_string acc r
+          | false, _, 0 -> go stack ~escaped ~quoted_string:(not quoted_string) (value :: acc) r
+          | false, false, _ -> go stack ~escaped ~quoted_string acc r
+          | false, true,  _ -> failwith "Should never happen")
```

Comment:
Better to replace it by an `assert false` (which gives you the line where it is raised) + a command :+1:.

Reply:
"a comment"